core/src/main/java/org/apache/accumulo/core/clientImpl/ThriftTransportPool.java

```diff
@@ -654,6 +655,31 @@ public void consumeBuffer(int len) {
       }
     }
 
+    @Override
+    public TConfiguration getConfiguration() {
+      return wrappedTransport.getConfiguration();
+    }
+
+    @Override
+    public void updateKnownMessageSize(long size) throws TTransportException {
+      try {
+        ioCount++;
```

Comment:
Why the double counting?

Reply:
No idea. But it has been there since the inception of the code. It doesn't make any sense to me.

Reply:
Might have something to do with `// when ioCount is odd, it means I/O is currently happening`. I don't know if it factors there but I can see the case to make it odd before an operation, and when done, make it even again.

Reply:
Ah I guess that makes sense then. We just have to decide if those new methods count as IO or not. I thought the 2 that were doing something with the message would count as IO, while just calling `getConfiguration()` did not.

Reply:
Realize the comments above explained this too. In checkForStuckIO()

```
// Tried to call System.currentTimeMillis() and Thread.currentThread() before every io
      // operation.... this dramatically slowed things down. So switched to
      // incrementing a counter before and after each io operation.

      if ((ioCount & 1) == 1) {
        // when ioCount is odd, it means I/O is currently happening
        if (ioCount == lastIoCount) {

```